src/hooks/useDevtoolsContext.ts

```diff
@@ -1,28 +1,47 @@
-import { useMemo, useRef } from 'react';
-
-/** Find the name of the calling component. */
-const getHookParent = () => {
-  /**
-   * Line values could be:
-   * - named export             " at exports.Home (Home.tsx?9b1b:17)"
-   * - default export / inline  " at Home (Home.tsx?9b1b:17)"
-   */
-  const stackLine = (new Error().stack as string).split('\n')[3];
-  const parsedLine = /.*at (exports\.)?(\w+)/.exec(stackLine);
-
-  return parsedLine === null || typeof parsedLine[2] !== 'string'
-    ? 'Unknown'
-    : parsedLine[2];
-};
+import * as React from 'react';
+import { OperationContext } from '../types';
+
+const {
+  ReactCurrentOwner: CurrentOwner,
+} = (React as any).__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
+
+// Is the Fiber a FunctionComponent, ClassComponent, or IndeterminateComponent
+const isComponentFiber = ({ tag }: { tag: number }) =>
+  tag === 0 || tag === 1 || tag === 2;
+
+// Is the component one of ours (just a heuristic to avoid circular dependencies or flags)
+const isInternalComponent = (Component: { name: string }) =>
+  Component.name === 'Query' ||
+  Component.name === 'Mutation' ||
+  Component.name === 'Subscription';
+
+const useDevtoolsContextImpl = (): Partial<OperationContext> => {
+  return React.useMemo(() => {
+    let source = 'Component';
+
+    // Check whether the CurrentOwner is set
+    const owner = CurrentOwner.current;
+    if (owner !== null && isComponentFiber(owner)) {
+      let Component = owner.type;
+
+      // If this is one of our own components then check the parent
+      if (isInternalComponent(Component) && isComponentFiber(owner.return)) {
+        Component = owner.return.type;
+      }
 
-const useDevtoolsContextHook = () => {
-  const source = useRef(getHookParent());
+      // Get the Component's name if it has one
+      if (typeof Component === 'function') {
+        source = Component.displayName || Component.name || source;
+      }
+    }
 
-  return useMemo(() => [{ meta: { source: source.current } }], []);
+    return { meta: { source } };
+  }, []);
 };
 
 /** Creates additional context values for serving metadata to devtools. */
-export const useDevtoolsContext =
-  process.env.NODE_ENV === 'production'
-    ? () => [undefined]
```

Comment:
I've changed the output not to be an array, since it doesn't need to be one :) this is an internal hook so we can always change it without breaking anyone's apps luckily :+1:

Reply:
A bit of a deviation from the standard`[value, mutation/s]` which we've also been following throughout the project so far (useQuery, useMutation, etc).

Not sure this change is necessary/beneficial.